### PR TITLE
User can login

### PR DESCRIPTION
**Pull request by kfrostare**
### User Story
```
As a user of CoPing
In order to access features
I would like to be able to login
```
## Changes proposed in this pull request:
* Adds login functionality
* Adds logout functionality
* Restrict- functionality based on state authenticated
* Updates new trip- tests with login

## What I have learned working on this feature:
* There is a lot we don't know about React Native (at the moment)

## Packages/Gems added
- J-tockauth
- Expo-linear-gradient

## Screenshots (Client)
image

## Lo-Fi's (Client)
image

### CODE REVIEW

File: cypress/integration/visitorCanLogin.feature.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    cy.get("#submit-login")
+      .contains("Submit")
+      .click();
+    cy.get("#login-message").should(
+      "contain",
+      "Welcome back Awesome Possumsson"
+    );
+    cy.get("#close-login-form").click();
+    cy.get("#email").should("not.exist");
+    //add test that unauthenticated user can not see new trip and after close they can
```
Comment (johanbounce) [inline cypress/integration/visitorCanLogin.feature.js]: oops

File: cypress/integration/visitorCanLogin.feature.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    //add test to logout and that new trip disappears
```
Comment (johanbounce) [inline cypress/integration/visitorCanLogin.feature.js]: oops

**Review by emtalen — COMMENTED** [review]
Just fix so the password shows as dots, otherwise great job! 💯

**Review by johanbounce — COMMENTED** [review]
When can I buy this app from App Store?

**Review by pgaunitz — APPROVED** [review]
NICE

**Pull request merged**